src/Integration.Vsix/SonarLintNotificationsPackage.cs

```diff
@@ -0,0 +1,121 @@
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters.Binary;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Integration.Notifications;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    [PackageRegistration(UseManagedResourcesOnly = true)]
+    [InstalledProductRegistration("#110", "#112", "1.0", IconResourceID = 400)] // Info on this package for Help/About
+    [Guid(PackageGuidString)]
+    [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1650:ElementDocumentationMustBeSpelledCorrectly", Justification = "pkgdef, VS and vsixmanifest are valid VS terms")]
+    [ProvideAutoLoad(UIContextGuids.NoSolution)]
+    [ProvideAutoLoad(UIContextGuids.SolutionExists)]
+    public sealed class SonarLintNotificationsPackage : Package
+    {
+        /// <summary>
+        /// SonarLintNotifications GUID string.
+        /// </summary>
+        public const string PackageGuidString = "c26b6802-dd9c-4a49-b8a5-0ad8ef04c579";
+
+        private const string NotificationDataKey = "NotificationEventData";
+        private IActiveSolutionBoundTracker activeSolutionBoundTracker;
+        private ISonarQubeNotifications notifications;
+        private readonly IFormatter formatter = new BinaryFormatter();
+        private NotificationData notificationData;
+
+        private NotificationIndicator notificationIcon;
+
+        protected override void Initialize()
+        {
+            AddOptionKey(NotificationDataKey);
+            base.Initialize();
+
+            notifications = this.GetMefService<ISonarQubeNotifications>();
+
+            activeSolutionBoundTracker = this.GetMefService<IActiveSolutionBoundTracker>();
+            activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionBindingChanged;
+        }
+
+        private void OnSolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs binding)
+        {
+            if (notificationIcon == null)
+            {
+                notificationIcon = new NotificationIndicator();
+                notificationIcon.DataContext = notifications.Model;
+                VisualStudioStatusBarHelper.AddStatusBarIcon(notificationIcon);
+            }
+
+            if (binding.IsBound)
+            {
+                notifications.StartAsync(binding.ProjectKey, notificationData);
+            }
+            else
+            {
+                notifications.Stop();
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+
+            if (notificationIcon != null)
+            {
+                VisualStudioStatusBarHelper.RemoveStatusBarIcon(notificationIcon);
+            }
+
+            activeSolutionBoundTracker.SolutionBindingChanged -= OnSolutionBindingChanged;
+
+            notifications.Dispose();
```

Comment:
Not sure whether VS can end up calling `Dispose` twice but I would make sure not to dispose notifications twice just in case.